### PR TITLE
Add Node.op() for backwards compatibility #4519

### PR DESCRIPTION
**Pull request by saulpw**

### CODE REVIEW

File: ibis/expr/operations/core.py [hunk 1]
```diff
@@ -44,6 +44,11 @@ def equals(self, other):
             )
         return self.__cached_equals__(other)
 
+    def _op(self):
```
Comment (kszucs) [inline ibis/expr/operations/core.py:47]: I think the following should work:

```py
@deprecated(version="4.0", instead="remove intermediate .op() calls")
def op(self):
    return self
```

Comment (kszucs) [reply ibis/expr/operations/core.py:47]: Also we need a test case for it.

Comment (saulpw) [reply ibis/expr/operations/core.py]: > I think the following should work:

Of course.  I don't know what I was thinking, thanks for catching it.

Comment (cpcloud) [reply ibis/expr/operations/core.py]: I don't think a test is necessary here. The current CI failures indicate the code is exercised, and we aren't going to keep this around forever. Enshrining the behavior with a test is undesirable IMO.

**Review by kszucs — APPROVED** [review]

**Review by kszucs — CHANGES_REQUESTED** [review]
Seems like it causes conflicts in some of the backends due to `hasattr(op, 'op')` checks.

**Comment by cpcloud** [comment]
Thanks for doing this!

**Pull request merged**